crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -363,6 +363,24 @@ impl<'src> Parser<'src> {
         }
 
         match self.current_token_kind() {
+            TokenKind::DoubleStar => {
+                self.bump(TokenKind::DoubleStar);
+                let right =
+                    self.parse_binary_expression_or_higher(OperatorPrecedence::Exponent, context);
+                return Expr::BinOp(ast::ExprBinOp {
+                    left: Box::new(Expr::Name(ast::ExprName {
+                        range: self.missing_node_range(),
+                        id: Name::empty(),
+                        ctx: ExprContext::Invalid,
+                        node_index: AtomicNodeIndex::NONE,
+                    })),
+                    op: Operator::Pow,
+                    right: Box::new(right.expr),
+                    range: self.node_range(start),
+                    node_index: AtomicNodeIndex::NONE,
+                })
+                .into();
```

Comment:
I'm not sure if this is the right place for the fix. Would it be possible to add it to `parse_binary_expression_or_higher` or even in `parse_binary_expression_or_higher_recursive`? If not, can you explain why?

Moving it their would give you the diagnostic for free